### PR TITLE
Updates for sonar development

### PR DESCRIPTION
**Pull request by jsonar-cpapke**

### CODE REVIEW

File: modules/aws/agentless-gw/main.tf [hunk 1]
```diff
@@ -42,19 +42,27 @@ module "gw_instance" {
...
   hub_sonarw_public_key             = var.hub_sonarw_public_key
   hadr_dr_node                      = var.hadr_dr_node
   main_node_sonarw_public_key       = var.main_node_sonarw_public_key
   main_node_sonarw_private_key      = var.main_node_sonarw_private_key
-  proxy_info                        = var.ingress_communication_via_proxy
+  proxy_info                        = var.gw_proxy_info
   skip_instance_health_verification = var.skip_instance_health_verification
   terraform_script_path_folder      = var.terraform_script_path_folder
-  use_public_ip                     = false
+  use_public_ip                     = var.use_public_ip
```
Comment (assaf13) [inline modules/aws/agentless-gw/main.tf:63]: If you expose use_public_ip, don't you need to expose also attach_persistent_public_ip ?

File: modules/aws/agentless-gw/main.tf [hunk 2]
```diff
@@ -42,19 +42,27 @@ module "gw_instance" {
...
+  sonargd_password_secret_name      = var.sonargd_password_secret_name
   ssh_key_path                      = var.ssh_key_pair.ssh_private_key_file_path
   binaries_location                 = var.binaries_location
+  tarball_url                       = var.tarball_url
   hub_sonarw_public_key             = var.hub_sonarw_public_key
   hadr_dr_node                      = var.hadr_dr_node
   main_node_sonarw_public_key       = var.main_node_sonarw_public_key
   main_node_sonarw_private_key      = var.main_node_sonarw_private_key
-  proxy_info                        = var.ingress_communication_via_proxy
+  proxy_info                        = var.gw_proxy_info
```
Comment (assaf13) [inline modules/aws/agentless-gw/main.tf:60]: If you want to change this variable name, can you also change it at the Azure modules as well. We tried to create the same API and flow for AWS and Azure modules

File: modules/aws/agentless-gw/variables.tf [hunk 3]
```diff
@@ -89,6 +95,11 @@ variable "public_ip" {
   description = "Create public IP for the instance"
 }
 
+variable "termination_protection" {
```
Comment (assaf13) [inline modules/aws/agentless-gw/variables.tf:98]: Please add description to this variable

File: modules/aws/hub/main.tf [hunk 4]
```diff
@@ -38,6 +38,12 @@ locals {
   ]
 }
 
+resource "tls_private_key" "sonarw_private_key" {
```
Comment (assaf13) [inline modules/aws/hub/main.tf]: Where will you use this resource ?

File: modules/aws/sonar-base-instance/setup.tftpl [hunk 5]
```diff
@@ -141,8 +149,13 @@ function attach_disk() {
...
     echo Downloading tarball..
     # Download installation tarball
-    TARBALL_FILE=$APPS_DIR/$(basename ${installation_s3_key})
-    /usr/local/bin/aws s3 cp s3://${installation_s3_bucket}/${installation_s3_key} $TARBALL_FILE --region ${installation_s3_region} >/dev/null
+    if [[ -n "${tarball_url}" ]]; then
+        TARBALL_FILE=$(basename "${tarball_url}")
+        wget "${tarball_url}" -O ./$TARBALL_FILE
+    else
+        TARBALL_FILE=$(basename ${installation_s3_key})
+        /usr/local/bin/aws s3 cp s3://${installation_s3_bucket}/${installation_s3_key} ./$TARBALL_FILE --region ${installation_s3_region} >/dev/null
```
Comment (assaf13) [inline modules/aws/sonar-base-instance/setup.tftpl]: Why did you change it to download the tarball to the current directory ?

File: modules/null/federation/grab_lock.sh [hunk 6]
```diff
@@ -10,14 +11,15 @@ while [ $lock_acquired -eq 0 ]; do
     while ls $lock_file_prefix* &>/dev/null; do
         rm -f $lock_file
         echo "waiting for previous federation to complete.."
-        sleep $(($RANDOM * 100 / 100000))
+        sleep "$(( ( RANDOM * 100 ) / 100000 ))"
```
Comment (assaf13) [inline modules/null/federation/grab_lock.sh:14]: In general, your lock file code seems more robust and accurate but since Eytan developed this code and he won't be at the office in the next few days, I don't feel comfortable to approve it. Did you test it ?

File: modules/null/federation/variables.tf [hunk 7]
```diff
@@ -49,3 +51,9 @@ variable "hub_proxy_info" {
     proxy_ssh_user             = null
   }
 }
+
+variable "use_public_ip" {
```
Comment (assaf13) [inline modules/null/federation/variables.tf]: I didn't see where you are using this variable at the federation module

File: modules/null/federation/main.tf [hunk 8]
```diff
@@ -5,18 +5,19 @@
 locals {
   lock_shell_cmds = file("${path.module}/grab_lock.sh")
   federate_hub_cmds = templatefile("${path.module}/federate_hub.tftpl", {
-    ssh_key_path                   = var.hub_info.hub_private_ssh_key_path
-    dsf_gw_ip                      = var.gw_info.gw_ip_address
-    dsf_hub_ip                     = var.hub_info.hub_ip_address
-    hub_ssh_user                   = var.hub_info.hub_ssh_user
+    ssh_key_path                   = var.hub_info.private_ssh_key_path
+    dsf_gw_ip                      = var.gw_info.federation_ip_address
```
Comment (assaf13) [inline modules/null/federation/main.tf]: dsf_gw_federation_ip ?

File: modules/aws/hub/variables.tf [hunk 9]
```diff
@@ -150,6 +155,12 @@ variable "binaries_location" {
   nullable    = false
 }
 
+variable "tarball_url" {
+  type        = string
+  default     = ""
+  description = "https DSF installation location, defaults to using binaries_location"
```
Comment (assaf13) [inline modules/aws/hub/variables.tf]: Please change the description to:
HTTPS DSF installation location. If not set, binaries_location is used

Change binaries_location description to:
S3 DSF installation location. If tarball_url not set, binaries_location is used

Change at all modules: hub, GW and base instance

File: modules/aws/agentless-gw/variables.tf [hunk 10]
```diff
@@ -124,6 +140,12 @@ variable "binaries_location" {
   nullable    = false
 }
 
+variable "tarball_url" {
```
Comment (assaf13) [inline modules/aws/agentless-gw/variables.tf:148]: Please change it to binaries_url in order to be consistent with binaries_location

File: modules/aws/sonar-base-instance/setup.tftpl (hunk 5 above)
Comment (jsonar-cpapke) [reply modules/aws/sonar-base-instance/setup.tftpl]: It would be downloaded to the current directory either way, but this was updated somewhat recently - https://github.com/imperva/dsfkit/pull/266/files#diff-5284ff956f8e3d6384e1c0a67deb292ac2f9a0d550dbdd21d4ac1fcbf4f76952

File: modules/aws/agentless-gw/main.tf (hunk 1 above)
Comment (jsonar-cpapke) [reply modules/aws/agentless-gw/main.tf:63]: Those would be two different things. This just allows me to run the job on my local machine, as it forces connection to the gateway over public IP. Whereas persistent IPs only make it static (but that doesn't affect the connection during setup).

File: modules/aws/hub/main.tf (hunk 4 above)
Comment (jsonar-cpapke) [reply modules/aws/hub/main.tf]: Not sure, I was just copying what Brian had done originally

File: modules/null/federation/grab_lock.sh (hunk 6 above)
Comment (jsonar-cpapke) [reply modules/null/federation/grab_lock.sh:14]: I did test it. Without the below change to the 0.2 second sleep, my federation froze trying to lock if there were 2 or more gateways.

File: modules/null/federation/main.tf (hunk 8 above)
Comment (jsonar-cpapke) [reply modules/null/federation/main.tf]: I'm not sure what you mean

File: modules/aws/agentless-gw/variables.tf (hunk 10 above)
Comment (assaf13) [reply modules/aws/agentless-gw/variables.tf:148]: Please ignore my last comment and keep it tarball_url

File: modules/aws/agentless-gw/main.tf (hunk 1 above)
Comment (assaf13) [reply modules/aws/agentless-gw/main.tf:63]: But if you already decide to provide an option to set a public IP to the GW, maybe at other use cases someone else would like to have the option to set also a persistent public IP

File: modules/null/federation/main.tf (hunk 8 above)
Comment (assaf13) [reply modules/null/federation/main.tf]: Change dsf_gw_ip variable here to dsf_gw_federation_ip

File: modules/null/federation/grab_lock.sh (hunk 6 above)
Comment (assaf13) [reply modules/null/federation/grab_lock.sh:14]: I will resolve it but let's test it once you merge this branch with the examples changes also to dev branch

File: modules/aws/agentless-gw/main.tf (hunk 1 above)
Comment (jsonar-cpapke) [reply modules/aws/agentless-gw/main.tf:63]: I'm still not quite sure what you mean, this is just related to what IP to use when running ssh.

File: modules/aws/agentless-gw/main.tf (hunk 2 above)
Comment (jsonar-cpapke) [reply modules/aws/agentless-gw/main.tf:60]: Do we want to change it? This was something Brian had changed, and I just copied over his code.

File: modules/aws/agentless-gw/variables.tf (hunk 3 above)
Comment (jsonar-cpapke) [reply modules/aws/agentless-gw/variables.tf:98]: Done

File: modules/aws/agentless-gw/main.tf (hunk 2 above)
Comment (assaf13) [reply modules/aws/agentless-gw/main.tf:60]: The name of proxy_info sounds better to me. But I will let you decide if to change it at all the places or to revert this change

Comment (jsonar-cpapke) [reply modules/aws/agentless-gw/main.tf:60]: Updated everything

File: modules/azurerm/hub/variables.tf [hunk 11]
```diff
@@ -124,11 +124,11 @@ variable "storage_details" {
   description = "Compute instance volume attributes"
 }
 
-variable "ingress_communication_via_proxy" {
+variable "wh_proxy_info" {
```
Comment (assaf13) [inline modules/azurerm/hub/variables.tf]: Please change it to hub_proxy_info

Comment (jsonar-cpapke) [reply modules/azurerm/hub/variables.tf]: Updated
